### PR TITLE
Store chainlink revocations

### PR DESCRIPTION
**Pull request by mlsteele**
This saves 5% of cpu time spent during a cold load of keybasefriends.

Is this safe? I'm not sure what the memory rules are around this code.

### CODE REVIEW

**Comment by mlsteele** [comment]
GetRevocations is not on the map anymore. So it can't shave another 9?

**Comment by mlsteele** [comment]
Bah it should've only halved. This doesn't make sense.

**Comment by mlsteele** [comment]
The expected result of this change would be for GetRevocations to halve in its cpu contribution. The svg output of pprof made it seem like it went from 10% to negligible. Hence my confusion. But `pprof -top` tells the truth that it went from 10% to 5%. I wonder why the svg did that.

**Comment by mlsteele** [comment]
Anyway now it's 4%

**Comment by maxtaco** [comment]
So that means it helped?

**Comment by mlsteele** [comment]
Yep!

**Comment by maxtaco** [comment]
cool!

File: go/libkb/chain_link.go [hunk 1]
```diff
@@ -198,16 +199,17 @@ var badWhitespaceChainLinks = map[keybase1.SigID]string{
...
+	id               LinkID
+	hashVerified     bool
+	sigVerified      bool
+	payloadVerified  bool
+	chainVerified    bool
+	storedLocally    bool
+	revoked          bool
+	unsigned         bool
+	dirty            bool
+	revocationsCache *[]keybase1.SigID
```
Comment (maxtaco) [inline go/libkb/chain_link.go]: could you just get away with `revocationsCache []keybase1.SigID` and then set it to be `[]keybase1.SigID{}` if caching an empty value?  might be slightly simpler.  I don't feel strongly.

**Pull request merged**